src/views/settings/Terminal/ComponentLog/ComponentLog.vue

```diff
@@ -84,6 +84,14 @@ const components_ee = [
   {
     'title': 'Xrdp',
     'components': ['xrdp', 'jms_xrdp']
+  },
+  {
+    'title': 'Nec',
+    'components': ['nec', 'jms_nec']
+  },
+  {
+    'title': 'Facelive',
+    'components': ['facelive', 'jms_facelive']
   }
 ]
 all_components.push(...components_ce)
```

Comment:
There are no major issues or irregularities found in this code snippet. However, there is one potential improvement that could optimize performance by reducing redundant component imports:
```cpp
if (components_ce.length <= 1 && !hasComponentImport(components_ce[0]))
        components_ce.pop();
```
This change reduces the number of import checks performed if only the first component is being used.

To further refine this code into optimal form, consider refactoring to use functions with clear names:

```cpp
add_component(const std::string& title);
add_components(const std::vector<std::pair<const char*, const char*>>& titles_and_components);

bool has_component(const std::string& component) { return components_.find(component) != components_.end(); }

void add_new_component(const std::string& component_title,
                      const std::map<std::string, ComponentInfo*>& info_map);

std::optional<Components> get_all_componenets() const;
```